projects/bim/index.js

```diff
@@ -0,0 +1,72 @@
+const utils = require('../helper/utils');
+const { toUSDTBalances } = require('../helper/balances');
+
+const apiUrl = 'https://staking-api.bim.finance'
+const vaultEndpointBase = 'https://raw.githubusercontent.com/bim-finance-org/staking-vaults/master/';
+const chainVaultEndpoints = {
+  10: vaultEndpointBase + "optimism.json",
+  100: vaultEndpointBase + "gnosis.json",
+  137: vaultEndpointBase + "polygon.json",
+  8453: vaultEndpointBase + "base.json",
+};
+
+const chains = {
+  xdai: 100,
+  polygon: 137,
+  base: 8453,
+  optimism: 10,
+}
+
+let lpsPrices;
+const getVaultBalances = async (chainId, vaults, api) => {
+  if (!vaults) {
+    throw new Error(`getVaultBalances: undefined vaults passed for ${chainId}`);
+  }
+  const calls = vaults.map(vault => {
+    return api.call({
+      abi: {
+        inputs: [],
+        name: 'balance',
+        outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+        stateMutability: 'view',
+        type: 'function',
+      },
+      target: vault.earnedTokenAddress,
+    });
+  });
+  const res = await Promise.all(calls);
+  return res;
+};
+
+function fetchChain(chainId) {
+  return async (_, _b, _cb, { api, }) => {
+    const chainVaults = utils.fetchURL(chainVaultEndpoints[chainId])
+    if (!lpsPrices) lpsPrices = utils.fetchURL(apiUrl + "/lps/breakdown")
+    const vaults = (await chainVaults).data;
+    const balances = await getVaultBalances(chainId, vaults, api);
+    const prices = (await lpsPrices).data;
+    let tvl = 0;
+    for (let i = 0; i < vaults.length; i++) {
+      const vault = vaults[i];
+
+      const vaultBalance = balances[i];
+      let tokenPrice = 0;
+      const price = prices[vault.id];
+      if (price) {
+        tokenPrice = price.price;
+      }
+      tvl += (vaultBalance / (10**(vault.tokenDecimals ?? 18))) * tokenPrice;
+    }
+
+    return toUSDTBalances(tvl);
```

Comment:
please export token balances instead. If there are any assets that we dont have pricing support for, please let us know and we will look at adding support for them

Reply:
What assets are the vaults holding? We should already support pricing for many defi assets and any others we should be able to add.